internal/mempool/v1/reactor.go

```diff
@@ -55,6 +55,8 @@ type Reactor struct {
 
 	mtx          tmsync.Mutex
 	peerRoutines map[types.NodeID]*tmsync.Closer
+
+	panicHandler func(interface{})
```

Comment:
It's often conventional to organize fields so that a mutex precedes the fields it protects. I don't know if we follow that convention here, but if so I think we should move this field up above `mtx`.

Also, could this field please get a brief comment?

Also, since this function doesn't really handle a panic, but rather acts as a notification receiver, I'd like to advocate against the suffix `Handler`. Perhaps: `logPanic` or `observePanic` as a name?

Reply:
From a brief glance over the methods in this file, it appears that `peerRoutines` is the field guarded by `mtx`. I like this convention and have seen it referred to as a `mutex hat`. 

Moved the function up, changed name, and commented.